cleanSweStore.pl

```diff
@@ -0,0 +1,169 @@
+my $dataToClean;
+while(<PROJECTS>){
+	if(!/^#/) {
+		chomp;
+		my ($runfolder,$project) = split(/\t/, $_);
+		$project =~ s/\s+//;
+		$dataToClean->{$runfolder}->{$project} = 1;
+	}
+}
+close(PROJECTS);
+
+my $timestamp = time;
+
+open REMOVED, "> removedFromSweStore.$timestamp.log" or die "Couldn't open output file: removedFromSweStore.$timestamp.log!\n";
+open SAVED, "> leftOnSweStore.$timestamp.log" or die "Couldn't open output file: leftOnSweStore.$timestamp.log!\n";
+
+print "Cleaning swestore!\n";
+#
+# Remove the provided projects from SweStore
+#
+foreach my $runfolder (keys %{$dataToClean}) { # Process each runfolder 
+	my ($year,$month,$day) = ($runfolder =~ m/^(\d{2})(\d{2})(\d{2})_[A-Z0-9]+_[0-9]+_[A-Z0-9]/);
+	# Find each  stored project at SweStore, for the specified runfolder
+	my $projects = qx(ils $swestorePath/20$year-$month/$runfolder/Projects/);
+	#Result from ils
+	#
+	# ils /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects/
+	# /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects:
+	# C- /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects/MK-0401
+	# C- /ssUppnexZone/proj/a2009002/2014-06/140605_D00118_0144_AC44G7ACXX/Projects/MK-0429
+	#
+	my @projectPath = split(/\n/,$projects);
+	my %foundProjects;
+	# Only extract information from lines containing "C-"
+	foreach (@projectPath) {
+		if(/^[ ]*C-/) {
+			my ($project) = ($_ =~ m/.*\/(.*)$/);
+			$project =~ s/\s+//;
+			$foundProjects{$project} = 1;
+		}
+	}
+	# Calculate number of projects found
+	my $numFoundProjects = (keys %foundProjects);
+	# Calculate number of projects that have been provided
+	my $numRemoveProjects = (keys %{$dataToClean->{$runfolder}});
+	#Make sure that the provided projects exist on SweStore, if not terminate the script.
+	foreach my $key (keys %{$dataToClean->{$runfolder}}) {
+		if(!exists($foundProjects{$key})) {
+			die "Couldn't find project $key on swestore: /ssUppnexZone/proj/a2009002/20" . $year . '-' . $month . '/' . $runfolder . "/Project\n";
+		}
+	}
+	
+	if($numFoundProjects < $numRemoveProjects) { #Cannot remove more projects than found on SweStore
+		die "You can't remove more projects than what exists, found $numFoundProjects, removing $numRemoveProjects!\n";
+	} elsif($numFoundProjects > $numRemoveProjects) { # Remove subset of projects found on SweStore 
+		if($execute) { #Perform deletion
+			foreach my $key (keys %{$dataToClean->{$runfolder}}) {
+				qx(irm -rf $swestorePath/20$year-$month/$runfolder/Projects/$key);
+				print REMOVED "$swestorePath/20$year-$month/$runfolder/Projects/$key\n";
+				delete $foundProjects{$key};
+			}
+			foreach my $key (keys %foundProjects) {
+				print SAVED "$swestorePath/20$year-$month/$runfolder/Projects/$key\n";
+			}
+		}
+		print("Removing a subset of the projects from $runfolder\n");
+	} else { # Same number of folders in file as on SweStore, remove entire runfolder
```

Comment:
actually, i'd suggest that we keep all data except for the "heavy" fastq files, even if all projects in the runfolder have been cleaned (see the email i sent yesterday)